azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/data/CallHistoryRepository.kt

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.calling.data
+
+import android.content.ContentValues
+import android.content.Context
+import com.azure.android.communication.ui.calling.data.model.CallHistoryRecordData
+import org.threeten.bp.LocalDateTime
+import org.threeten.bp.format.DateTimeFormatter
+
+internal interface CallHistoryRepository {
+    fun insert(callId: String, callDateTime: LocalDateTime)
+    fun getAll(): List<CallHistoryRecordData>
+    fun remove(id: Int)
+}
+
+internal class CallHistoryRepositoryImpl(
+    private val context: Context
+) : CallHistoryRepository {
+
+    private val datePattern: String = "yyyy-MM-dd HH:mm:ss"
+
+    override fun insert(callId: String, callDateTime: LocalDateTime) {
+        val db = DbHelper(context).writableDatabase
+
+        val date = callDateTime.format(DateTimeFormatter.ofPattern(datePattern))
+
+        val values = ContentValues().apply {
+            put(CallHistoryContract.COLUMN_NAME_CALL_ID, callId)
+            put(CallHistoryContract.COLUMN_NAME_CALL_DATE, date)
+        }
+
+        db.insert(CallHistoryContract.TABLE_NAME, null, values)
```

Comment:
Do the cleanup here as a single DELETE statement, wrapped in a transaction together with the INSERT.

Reply:
Done